### ISSUE TITLE
[Bug] conv_soft_argmax2d input range ?

### ISSUE THREAD

**Issue by sfikas** [kornia/kornia#382]
It seems that conv_soft_argmax2d should accept ranges only in 0.0 - 1.0. Otherwise 'NaN' values tend to appear, as per the following example:
![kornia_convsoft_error](https://user-images.githubusercontent.com/16489490/71200670-2c81bb00-22a1-11ea-8d36-fa1a26ae0c7d.jpg)

Is this intended behaviour? If not, this is a bug -- If yes, it should be noted on the documentation and a check should also be added in the code, outputing an error if the input range and type is not proper (I am willing to submit the related pull request if this is the case).

**Comment by edgarriba** [comment]
@sfikas I recently found a similar behavior that could come from the same issue. Currently, inside the conv_softmax operator we compute the softmax with a handmade implementation defined as follows:

```x_exp = (input / temperature).exp()```

See: https://github.com/kornia/kornia/blob/master/kornia/geometry/spatial_soft_argmax.py#L391

However, it turns out that depending on the input data range, the exponential operator con explode or overflow. This could be the case when your previous layer is e.g. a ReLU. For this reason, there is this exponential normalization trick that will make softmax numerically more stable.

See: https://timvieira.github.io/blog/post/2014/02/11/exp-normalize-trick/

In fact, if you dig deeper into PyTorch itself, you will see that they already apply this trick which is totally transparent to the user.

https://github.com/pytorch/pytorch/blob/bcb0bb7e0e03b386ad837015faba6b4b16e3bfb9/aten/src/ATen/native/SoftMax.cpp#L44

I will fix this issue myself during the day.

**Comment by sfikas** [comment]
Thanks! By the way, same goes for conv_soft_argmax3d, which was what I originally wanted to use, and this in turn because it is part of the affine-invariant keypoint detection tools in kornia.feature (class ScaleSpaceDetector etc.)

**Comment by sfikas** [comment]
The 'exploding' behaviour of this softmax implementation could (??) be related to the problem reported in https://github.com/kornia/kornia/issues/353  , @pablovela5620

**Comment by edgarriba** [comment]
probably. `conv_soft_argmax3d` is also affected ;D
